sprint_04

FilTon87 opened the pull request:
Задание спринт 04 для ревью

FilTon87 committed:
Верстка интерфейса и выводим первый вопрос на экран

FilTon87 committed:
Обрабатываем ответ пользователя

FilTon87 committed:
Показываем следующий вопрос
upd: при показе нового вопроса красим рамку в белый цвет

FilTon87 committed:
финал

FilTon87 committed:
финальные правки

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,9 +1,115 @@
+    }
+    
     // MARK: - Lifecycle
     override func viewDidLoad() {
```
Стоит организовывать код следующим образом:

- Публичные переменные
- IBOutlet
- Приватные переменные
- Публичные методы(сначала `init`ы, `override`ы, и потом остальные)
- Приватные методы
- IBAction

Подробнее почитать можно тут: [Организация кода](https://github.com/RedMadRobot/RMR-swift-style-guide#code-organization)

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,9 +1,115 @@
+    private func showAnswerResult(isCorrect: Bool) {
+        imageView.layer.masksToBounds = true // даём разрешение на рисование рамки
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
```
Сейчас при старте у постера нет закругления углов. Это происходит из-за того, что закругление не задается сразу как загрузилась view. Т.е закругление нужно делать один раз во `viewDidLoad`

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,9 +1,115 @@
+    }
+    
+    private func showNextQuestionOrResults() {
+        imageView.layer.borderColor = UIColor.ypWhite.cgColor
```
В макете нет белой рамки

helecta commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,9 +1,115 @@
+    private func showNextQuestionOrResults() {
+        imageView.layer.borderColor = UIColor.ypWhite.cgColor
+        if currentQuestionIndex == questions.count - 1 {
+            show(quizresult: QuizResultsViewModel(title: "Этот раунд окончен!", text: "Ваш результат \(correctAnswers)/10", buttonText: "Сыграть еще раз"))
```
Лучше такие длинные методы разбивать строчками:
```
            show(quizresult: QuizResultsViewModel(
                title: "Этот раунд окончен!",
                text: "Ваш результат \(correctAnswers)/10",
                buttonText: "Сыграть еще раз"))
```
Так код значительно легче и удобней читать

FilTon87 committed:
Изменения по итогам ревью:
- добавлено закругление углов у постера при старте;
- убрана белая рамка на постере после ответа;
- изменен цвет статус бара на белый;
- добавлена блокировка кнопок ответа "Да" и "Нет" на время обработки ответа до начала показа нового вопроса.
